### ISSUE TITLE
Blocked scheduler items intermittently not getting reschedule

### ISSUE THREAD

**Issue by warunalakshitha** [ballerina-platform/ballerina-lang#33827]
**Description:**
In the Jballerina runtime scheduler, we have scheduler groups and group items. The scheduler group contains a stack of scheduler items.

As an example, if you have 3 workers (except strand annotation with any) those will be added to the same scheduler group.

Each scheduler group is added to the runnable queue. Each os thread in scheduler there pool will assign a group( removed from runnable queue) and execute items.

Once execute each item can be done or blocked on another event like i/o operation, sleep. Then we will execute the next item in the group. If all items are done or blocked we will mark the group "scheduled" flag to be false. Then os thread will assign another group.

Same time blocked items can be done and once done the i/o thread will reschedule the given item. First, it will check the items' group "scheduled" flag and if it is still true means the given group is still running another thread. So we just added to the group items stack so the currently running scheduler thread will execute the item continue the next instructions.

If the group is already done ("scheduled flag false) i/o thread will add that group to the runnable queue.

The group scheduled status change is not an atomic operation. Hence there is a possibility i/o thread might think it is already an executable state and same time scheduler thread change the status to false, so that item gets lost without rescheduling.

So the item will be not done. This will make either program stuck or service requests time out.
